include/wasp/text/types.h

```diff
@@ -625,7 +661,7 @@ struct Memory {
   MemoryDesc desc;
   OptAt<InlineImport> import;
   InlineExportList exports;
-  optional<TextList> data;
+  optional<DataItemList> data;
```

Comment:
Why not just use the empty list here to signify no items?

Reply:
I've been trying to use `optional` for things like this, rather than sentinel values. It makes it nicer to do a quick "does this exist" check too (`data.has_value()`) -- otherwise you'd probably need a separate function.